### DISCUSSION TITLE
"java.lang.IllegalArgumentException: Cannot find the ODA dataSource extension..." critical error with new BIRT 4.11 on JDK/JRE 11

### DISCUSSION THREAD

**Discussion by ivkina** [eclipse-birt/birt#1152] [Q&A]
We have custom created Datasource/Dataset ODA extension that fails on latest BIRT 4.11 build. 
Steps:
1) download and install birt-report-designer-all-in-one-4.11.0-20221120-win32.win32.x86_64.zip from https://download.eclipse.org/birt/downloads/drops/latest/. E.g., c:\eclipse_411.
2) run eclipse/birt designer using jre 11, e.g., C:\eclipse_411\eclipse.exe -vm "C:\oracle\jdk-11.0.15\bin\javaw.exe"
3) open workspace "oda" and export/build 2 plugins: com.deltek.enterprise.system.reporting.client.birt.oda.designer_1.0.0.jar and com.deltek.enterprise.system.reporting.client.birt.oda.runtime_1.0.0.jar. 
Workspace "oda" attached - 
[eclipse_411_workspace.zip](https://github.com/eclipse/birt/files/10105781/eclipse_411_workspace.zip)

Built plugins also attached - 
[plugins.zip](https://github.com/eclipse/birt/files/10105787/plugins.zip)
4) copy 2 new plugins under your eclipse home/plugins folder, e.g., c:\eclipse_411\plugins
5) open workspace "reports" (attached in #3) and open syrusr.rptdesign attached - 
[rpt_lib.zip](https://github.com/eclipse/birt/files/10105827/rpt_lib.zip).
Note that rptdesign file references libraries (also attached above).
6) Go to report outline->DataSources and DataSets and note that CP_DataSource and DataSet are in red error state. Double click on DataSet and then click on "Property Bindings" -->> error occurs:
java.lang.IllegalArgumentException: Cannot find the ODA dataSource extension (com.deltek.enterprise.system.reporting.client.birt.oda.runtime).
 Check the workspace log file for any problems with loading the extension bundle and its dependencies.
	at org.eclipse.datatools.connectivity.oda.util.manifest.ManifestExplorer.getExtensionManifest(ManifestExplorer.java:200)....
Complete error attached - 
[error.txt](https://github.com/eclipse/birt/files/10105884/error.txt)

The same steps, same ODA extension/plugins are working just fine with BIRT 4.9 released on 4/2022.
Thanks,
Aleksey
Deltek Inc.

**Comment by speckyspooky** [comment]
I figured out the reason of your issue and you can fix it by your self.
**But currently I cannot say why this effect happens.**

I have the same issue with BIRT_4.8 an plugin-files which are built only with JDK8 without compatibility mode of other JDKs or of >= JDK11.
I see in your MANIFEST.MF that you have the same environment issue:

![plugin-MANIFEST-JDK1 8](https://user-images.githubusercontent.com/41593722/204366859-eafad939-2c25-412b-82a6-842f7b4f86a9.png)

The issue happens because due to the JSE1.8 the "org.eclipse.equinox.simpleconfigurator" doesn't create the bundle.info correctly.

So you have to add 2 lines manuelly with the according path (changed to your path):

01) path: C:\BIRT_4.11\eclipse\configuration\org.eclipse.equinox.simpleconfigurator

02) add the 2 plugin lines
com.deltek.enterprise.system.reporting.client.birt.oda.designer,1.0.0,plugins/com.deltek.enterprise.system.reporting.client.birt.oda.designer_1.0.0.jar,4,false
com.deltek.enterprise.system.reporting.client.birt.oda.runtime,1.0.0,plugins/com.deltek.enterprise.system.reporting.client.birt.oda.runtime_1.0.0.jar,4,false


Result - your data source without "red" cross:

![2022-11-28 20_46_44-C__BIRT_4 11_eclipse_configuration_org eclipse equinox simpleconfigurator_bundle](https://user-images.githubusercontent.com/41593722/204367893-729d33ab-15e5-46c5-bd69-356607871c80.png)

@all:
Does anyone know why the equinox.simpleconfigurator have such behavior?
Which step or configuration is to set that the bundle.info will be created automatically?

**Comment by ivkina** [comment]
That worked! I would never guess it to edit this "secret" equinox file....:)))
Thanks!!!
